### PR TITLE
Refactor AL script

### PR DESCRIPTION
**Pull request by y0ast**
- Add some flags to make it easier to use in experimentation
- Fix a bug with density code path not working with 0 initial train size
- Refactor outer while loop

cc @BlackHC

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -244,6 +244,14 @@ The following papers have used code from Uncertainty Baselines:
 
 ## Contributing
 
+### Formatting Code
```
Comment (zi-w) [inline README.md:247]: High level question: why is this script located under jft/?

File: baselines/jft/active_learning.py [hunk 2]
```diff
@@ -659,65 +680,70 @@ def main(config, output_dir, acquisition_method):
...
-        rng=train_ds_rng,
-        process_batch_size=local_batch_size,
-        preprocess_fn=preprocess_spec.parse(
-            spec=config.pp_eval, available_ops=preprocess_utils.all_ops()),
-        shuffle=False,
-        drop_remainder=False,
-        prefetch_size=config.get("prefetch_to_host", 2),
-        num_epochs=1,
-    )
+      repeated_train_ds = input_utils.get_data(
```
Comment (zi-w) [inline baselines/jft/active_learning.py:703]: what is `repeated_train_ds` and its difference to `train_eval_ds`? Please add some comments for context.

File: baselines/jft/active_learning.py [hunk 3]
```diff
@@ -659,65 +680,70 @@ def main(config, output_dir, acquisition_method):
...
-        process_batch_size=local_batch_size,
-        preprocess_fn=preprocess_spec.parse(
-            spec=config.pp_train, available_ops=preprocess_utils.all_ops()),
-        shuffle_buffer_size=config.shuffle_buffer_size,
-        prefetch_size=config.get("prefetch_to_host", 2),
-        # TODO(joost,andreas): double check if below leads to bootstrap sampling
-        num_epochs=num_repeats,
-    )
+    # Only fine-tune if there is anything to fine-tune with.
+    if current_train_ds_length > 0:
```
Comment (zi-w) [inline baselines/jft/active_learning.py:697]: It's difficult to tell in github formatting to see where this if ends. What happens for `current_train_ds_length` = 0?

File: baselines/jft/active_learning.py [hunk 4]
```diff
@@ -741,24 +767,28 @@ def main(config, output_dir, acquisition_method):
...
-          num_epochs=1)
-
-      pool_scores = get_density_scores(
-          model=model,
-          opt_repl=current_opt_repl,
-          train_ds=feature_train_ds,
-          pool_pre_logits=pool_outputs,
-          pool_masks=pool_masks)
+      if current_train_ds_length > 0:
+        feature_train_ds = input_utils.get_data(
```
Comment (zi-w) [inline baselines/jft/active_learning.py:771]: What's the difference of this `feature_train_ds` from the previous datasets generated by input_utils.get_data?

**Review by zi-w — APPROVED** [review]
Thank you Joost! Please address the comments or leave TODOs. I will start the merging soon.

File: baselines/jft/active_learning.py [hunk 5]
```diff
@@ -57,18 +58,29 @@
...
     "acquisition_method",
     default="uniform",
     enum_values=["uniform", "density", "margin", "entropy"],
     help="Choose an acquisition method.")
+flags.DEFINE_integer(
+    "acquisition_batch_size",
+    default=None,
+    help="Acquisition batch size per active learning iteration.")
+flags.DEFINE_integer(
+    "max_training_set_size", default=None, help="Maximum training set size.")
```
[the remark this answers is not in the document]
Comment (y0ast) [reply baselines/jft/active_learning.py:72]: It includes the initial training set, so for max_training_size = 200 and initial_size = 20, we acquire 180 points more.

We acquire using acquisition_batch_size until we reached training_set_size >= max_training_set_size. There's no specific constraint, but there is the possibility of going over the max size for 1 step if your equation doesn't hold.

File: baselines/jft/active_learning.py [hunk 6]
```diff
@@ -57,18 +58,29 @@
...
+flags.DEFINE_integer(
+    "initial_training_set_size",
+    default=None,
+    help="Initial training set size.")
```
[the remark this answers is not in the document]
Comment (y0ast) [reply baselines/jft/active_learning.py:76]: Yes, the initial set is randomly generated and yes, everything is controlled by FLAGS.seed including the initial set.

See from: https://github.com/google/uncertainty-baselines/pull/686/files#diff-17c1dccde3087d1a874c7d3cfbcb9fec3c1a5a6615c024cb5655aba8728b1bd2R653

File: README.md (hunk 1 above)
Comment (y0ast) [reply README.md:247]: We depend on code that is in this folder, so we need to be in this folder if we want to import it (it's not a package).

Comment (dustinvtran) [reply README.md:247]: For now, we're putting all code that leverages pretraining/finetuning with JFT/I21K models in jft/. I think this makes sense for now as there's a shared utilities required for such pretraining/finetuning.

File: baselines/jft/active_learning.py (hunk 3 above)
Comment (y0ast) [reply baselines/jft/active_learning.py:697]: We go directly to line 754 and evaluate the test accuracy.

This big if is not very pretty code and I'll look into refactoring it later.

File: baselines/jft/active_learning.py (hunk 2 above)
Comment (y0ast) [reply baselines/jft/active_learning.py:707]: See added comments in code.

**Review by zi-w — APPROVED** [review]
